File: JavaFx/Interface1/src/BubbleSort.java
/*
Bubble sort
 */
public class BubbleSort {

  public static void bubbleSort(int[] arr, int n) {
    for (int i = 0; i < n; i++) {
      boolean swapped = false;
      for (int j = 0; j < n - i - 1; j++) {
        if (arr[j] > arr[j + 1]) {
          int temp = arr[j];
          arr[j] = arr[j + 1];
          arr[j + 1] = temp;
          swapped = true;
        }
      }
      if (!swapped) {
        break;
      }
    }
  }

  public static void main(String[] args) {

    int arr[] = {45, 20, 50, 15, 10};
    System.out.print("Initial Array: ");
    for (int i : arr) {
      System.out.print(i + " ");
    }

    int n = arr.length;
    bubbleSort(arr, n); // called function
    System.out.print("\nSorted Array: ");
    for (int i : arr) {
      System.out.print(i + " ");
    }
  }
}

File: VS code/CP HANDBOOK/SORTING/bubblesort.cpp
// bubble sort
// O(n2)

#include <bits/stdc++.h>
using namespace std;

#define ll long long
#define nl endl

int main()
{
    ll arr[] = {2, 3, 6, 5, 7, 1, 4};
    ll n = 7;
    for (ll i = 0; i < n; i++)
    {
        for (ll j = 0; j < n - 1 - i; j++)
        {
            if (arr[j] > arr[j + 1])
            {
                swap(arr[j], arr[j + 1]);
            }
        }
    }
    for (ll i = 0; i < n; i++)
    {
        cout << arr[i] << " ";
    }
    cout << nl;

    return 0;
}